agrona/src/main/java/org/agrona/AsciiEncoding.java

```diff
@@ -379,6 +738,163 @@ public static int parseEightDigitsLittleEndian(final long bytes)
         return (int)val;
     }
 
+    /**
+     * Computes number of bits in a power of five number, i.e. {@code 5^power}
+     *
+     * @param power to which five is raised.
+     * @return number of bits in a number that be computed by raising five to the given {@code power}.
+     */
+    public static int powerOfFiveBitCount(final int power)
+    {
+        return ((power * 1217359) >>> 19) + 1;
+    }
+
+    /**
+     * Determines if the given value is multiple to the given power of five, i.e. if the {@code value} is divisible
+     * by {@code 5^power}.
+     *
+     * @param value to check.
+     * @param power of five factor.s
+     * @return {@code true} if value is divisible by the given power of five.
+     */
+    public static boolean isMultipleOfPowerOfFive(final long value, final int power)
+    {
+        // FIXME: Avoid modulo here
```

Comment:
Is there an easy solution to this?

Reply:
Yes, you by doing what is outlined in this [Faster Remainder by Direct Computation: Applications to Compilers and Software Libraries](https://arxiv.org/abs/1902.01961) paper.